barretenberg/cpp/src/barretenberg/ultra_vanilla_client_ivc/ultra_vanilla_client_ivc.cpp

```diff
@@ -0,0 +1,91 @@
+#include "barretenberg/ultra_vanilla_client_ivc/ultra_vanilla_client_ivc.hpp"
+#include "barretenberg/ultra_honk/oink_prover.hpp"
+
+namespace bb {
+
+void UltraVanillaClientIVC::accumulate(Circuit& circuit, const Proof& proof, const std::shared_ptr<VK>& vk)
+{
+    RecursiveVerifier verifier{ &circuit, std::make_shared<RecursiveVK>(&circuit, vk) };
+    Accumulator agg_obj = stdlib::recursion::init_default_aggregation_state<Circuit, stdlib::bn254<Circuit>>(circuit);
+    accumulator = verifier.verify_proof(proof, agg_obj).agg_obj;
+}
+
+HonkProof UltraVanillaClientIVC::prove(CircuitSource<Flavor>& source, const bool cache_vks)
+{
+    for (size_t step = 0; step < source.num_circuits(); step++) {
```

Comment:
Is this CircuitSource interface meant only for testing or are you imagining that you'd have some version of the class that would be loaded up with acir bytecode in actual usage?